modules/clickhouse/clickhouse_test.go

```diff
@@ -0,0 +1,274 @@
+	// }
+
+	conn, err := ch.Open(&ch.Options{
+		Addr: []string{connectionHost},
+		Auth: ch.Auth{
+			Database: dbname,
+			Username: user,
+			Password: password,
+		},
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, conn)
+	defer conn.Close()
+
+	// perform assertions
+	err = conn.Exec(context.Background(), createTableQuery)
+	assert.NoError(t, err)
+
+	err = conn.Exec(context.Background(), insertQuery)
+	assert.NoError(t, err)
+
+	rows, err := conn.Query(context.Background(), selectQuery)
+	assert.NoError(t, err)
+	assert.NotNil(t, rows)
+
+	var data []Test
+	for rows.Next() {
+		var r Test
+		err := rows.Scan(&r.Id)
+
+		assert.NoError(t, err)
+
+		data = append(data, r)
+	}
+
+	assert.Len(t, data, 1)
+}
+
+func TestClickHouseDSN(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := RunContainer(ctx, WithUsername(user), WithPassword(password), WithDatabase(dbname))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		assert.NoError(t, container.Terminate(ctx))
+	})
+
+	// connectionString {
+	connectionString, err := container.ConnectionString(ctx, "debug=true")
+	assert.NoError(t, err)
+	// }
+
+	opts, err := ch.ParseDSN(connectionString)
+	assert.NoError(t, err)
+
+	conn, err := ch.Open(opts)
+	assert.NoError(t, err)
+	assert.NotNil(t, conn)
+	defer conn.Close()
+
+	// perform assertions
+	err = conn.Exec(context.Background(), createTableQuery)
+	assert.NoError(t, err)
+
+	err = conn.Exec(context.Background(), insertQuery)
+	assert.NoError(t, err)
+
+	rows, err := conn.Query(context.Background(), selectQuery)
+	assert.NoError(t, err)
+	assert.NotNil(t, rows)
+
+	var data []Test
+	for rows.Next() {
+		var r Test
+		err := rows.Scan(&r.Id)
+
+		assert.NoError(t, err)
+
+		data = append(data, r)
+	}
+
+	assert.Len(t, data, 1)
+}
+
+func TestClickHouseWithInitScripts(t *testing.T) {
+	ctx := context.Background()
+
+	// withInitScripts {
+	container, err := RunContainer(ctx,
+		WithUsername(user),
+		WithPassword(password),
+		WithDatabase(dbname),
+		WithInitScripts(filepath.Join("testdata", "init-db.sh")),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// }
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		assert.NoError(t, container.Terminate(ctx))
+	})
+
+	connectionHost, err := container.ConnectionHost(ctx)
+	assert.NoError(t, err)
+
+	conn, err := ch.Open(&ch.Options{
+		Addr: []string{connectionHost},
+		Auth: ch.Auth{
+			Database: dbname,
+			Username: user,
+			Password: password,
+		},
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, conn)
+	defer conn.Close()
+
+	// perform assertions
+	rows, err := conn.Query(context.Background(), selectQuery)
+	assert.NoError(t, err)
+	assert.NotNil(t, rows)
+
+	var data []Test
+	for rows.Next() {
+		var r Test
+		err := rows.Scan(&r.Id)
+
+		assert.NoError(t, err)
+
+		data = append(data, r)
+	}
+
+	assert.Len(t, data, 1)
+}
+
+func TestClickHouseWithConfigFile(t *testing.T) {
```

Comment:
I'm missing a test with the YAML config file, as it's exposed in the API. The YAML file would be added to the testdata dir, and I'd also recommend (to avoid much duplication) to enclose the exec/query code block into a function that would be called else where. Thoughts?

Reply:
To be honest, I thought that if it works with `config.xml`, it would probably work with YAML too 😆 

I would add it, not problem.

I am aware about duplications but like I said in my first comment I left them on purpose. But I will work on it if it seems noisy